### PR TITLE
Update ec2_scaling_policy to use boto3 for python3.

### PR DESCRIPTION
**Pull request by actown**
##### SUMMARY
Upgrades ec2_scaling_policy to use boto3 for python3.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ec2_scaling_policy

##### ADDITIONAL INFORMATION
I am currently using this version of the module in production without issues.

### CODE REVIEW

File: plugins/modules/ec2_scaling_policy.py [hunk 1]
```diff
@@ -91,97 +93,108 @@ def create_scaling_policy(connection, module):
...
+            policy['MinAdjustmentStep'] = min_adjustment_step
 
         # check the remaining attributes
-        for attr in ('adjustment_type', 'scaling_adjustment', 'cooldown'):
-            if getattr(policy, attr) != module.params.get(attr):
+        for attr in ('AdjustmentType', 'ScalingAdjustment', 'Cooldown'):
+            if policy.get(attr) != module.params.get(attr):
                 changed = True
-                setattr(policy, attr, module.params.get(attr))
+                policy[attr] = module.params.get(attr)
```
Comment (flowerysong) [inline plugins/modules/ec2_scaling_policy.py:151]: This looks wrong; module parameters are snake_case, but attr is now CamelCase.

Comment (actown) [reply plugins/modules/ec2_scaling_policy.py:151]: It looks like I might be able to use something like snake_dict_to_camel_dict as seen here https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/ec2_lc.py#L541. I'll give that a try.

**Comment by actown** [comment]
> @actown Thanks for your work there's also some nice clean-up in here.
> 
> The migration to boto3 is a pretty significant change, it would be helpful if you could include some integration tests in your PR. We have some initial documentation over on the amazon.aws repo https://github.com/ansible-collections/amazon.aws/blob/main/CONTRIBUTING.md#pull-requests
> 
> One minor nit: rather than just calling `fail_json_aws(e)` it's more user friendly to also add an explanation of what was happening when the exception failed. `fail_json_aws(e, "Failed to remove scaling policy")`

I'll see if I can get any tests in, but in the meantime I have updated the fail_json_aws calls on my local copy.

**Comment by actown** [comment]
> > @actown Thanks for your work there's also some nice clean-up in here.
> > The migration to boto3 is a pretty significant change, it would be helpful if you could include some integration tests in your PR. We have some initial documentation over on the amazon.aws repo https://github.com/ansible-collections/amazon.aws/blob/main/CONTRIBUTING.md#pull-requests
> > One minor nit: rather than just calling `fail_json_aws(e)` it's more user friendly to also add an explanation of what was happening when the exception failed. `fail_json_aws(e, "Failed to remove scaling policy")`
> 
> I'll see if I can get any tests in, but in the meantime I have updated the fail_json_aws calls on my local copy.

I've pushed up a copy with the improved fail_json_aws calls.

File: plugins/modules/ec2_scaling_policy.py (hunk 1 above)
Comment (actown) [reply plugins/modules/ec2_scaling_policy.py:151]: I believe I have a change for this, once I'm able to test it I can commit it to this branch for review.

**Comment by tremble** [comment]
Ergh, I just noticed that you still had this one open, and I ported over another old PR which includes tests:

Would you be willing to take a look at:
https://github.com/ansible-collections/community.aws/pull/197

and see if that works for you?

**Comment by actown** [comment]
Feel free to close with the merge of #197

**Pull request closed without merging**